controller/controller.mustache

```diff
@@ -64,13 +65,22 @@ module.exports = {
   create: ({{name}}) => {
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
+    {{name}}.rev = uuid.v4();
     return r.table('{{pluralName}}').insert({{name}}, {returnChanges: true}).run()
     .then(rethinkdb.firstChange);
   },
   update: ({{name}}) => {
+    if (!{{name}}.rev) return Promise.reject(new httpError('BAD_REQUEST', 'No rev provided'));
     const valid = schema.validate({{name}});
     if (!valid) throw new httpError('UNPROCESSABLE_ENTITY');
-    return r.table('{{pluralName}}').update({{name}}, {returnChanges: true}).run()
+    return r.table('{{pluralName}}')
+    .get({{name}}.id)
+    .update(
+      rethinkdb.ifRevMatches({{name}}),
+      {returnChanges: true}
+    )
+    .run()
+    .then(rethinkdb.checkRevError)
```

Comment:
> It does support a .catch, but the driver doesn't actually throw an error in response to a Rethink error. It just returns an object with a first_error property containing the error message. Kinda odd behaviour from the driver, but hey ho.

Nya, bummer.

> "I don't know if this thing exists or not. If it doesn't, create it. If it does, but it looks different than the way I thought it might, bail. If it does exist and it looks like the way I thought it might, update it"

Yeah, I agree with that, though I like the readability of an upsert better.  There's more happening under the hood, with less code on top, which I think we should try and lean towards for redbeard in general, but what you're saying makes sense so yep let's leave it.